src/librustc_typeck/outlives/mod.rs

```diff
@@ -23,7 +27,51 @@ pub fn provide(providers: &mut Providers) {
 }
 
 //todo
-fn inferred_outlives_of<'a, 'tcx>(_tcx: TyCtxt<'a, 'tcx, 'tcx>, _def_id: DefId)
+fn inferred_outlives_of<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, def_id: DefId)
                                   -> Vec<ty::Predicate<'tcx>> {
-    Vec::new()
+    // Assert that this is a local node-id
+    let node_id = tcx.hir.as_local_node_id(def_id).unwrap();
+
+    match tcx.hir.get(node_id) {
+        hir_map::NodeItem(item) => match item.node {
+            hir::ItemStruct(..) => Vec::new(),
+            hir::ItemEnum(..) => Vec::new(),
+            _ => Vec::new(),
+        },
+        _ => Vec::new(),
+    }
+}
+
+#[allow(dead_code)]
+fn inferred_outlives_crate <'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>, def_id: DefId)
```

Comment:
What is the `def_id` parameter here supposed to refer to? My thought was that this function would infer the `outlives` relationships for **all structs/enums in the crate**, therefore it probably doesn't need to take a `def_id` (rather, in the body it will loop over the set of items to find those structs/enums).

Reply:
got it. yes I had a little mis-understanding about the function.

Reply:
btw planning to use `tcx.hir.krate().visit_all_item_likes(&mut terms_cx)` along with `impl ItemLikeVisitor` as we do in variance.

Reply:
sg, this function should presumably take a `CrateNum` then -- at least, that's what we've done elsewhere. (I might be inclined to change the convention for singleton queries like this, but let's not do it in this PR.)